index.bs

```diff
@@ -77,6 +77,30 @@ The module in this new sidebar context will never need to reflow to the wider la
 	<figcaption>A stylesheet heatmap showing added redundancy and rewriting of existing styles required to accommodate the layout variation in <a href="#fig-4">fig. 4</a> </figcaption>
 </figure>
 
+<h2 id="limitations-of-current-techniques">Limitations of current techniques</h2>
+
+There is currently no native CSS mechanism available to developers that achieves the primary function of container queries, which is to <em>employ the full capabilities of CSS to a specific element tree based on the state and properties of their container</em>. Therefore developers must look to workarounds to achieve their goals. These workarounds and their limitations are outlined here:
+
+The use of <strong>Media Queries</strong> can only affect CSS changes based on the state and properties of the viewport. The viewport is not a good indicator of the state and properties of a particular container within the viewport.
+
+The use of <strong>CSS layout methods</strong> such as <strong>Flexbox</strong> and <strong>Grid</strong>, alongside <strong>CSS-based workarounds</strong> (such as the <a href="https://medium.freecodecamp.org/the-fab-four-technique-to-create-responsive-emails-without-media-queries-baf11fdfa848">Fab Four technique</a>) cannot employ the full capabilities of CSS on an element tree.
+
+The use of <strong>iframes</strong> produces the closest effect to what Container Queries is intended to achieve, but their use has three significant drawbacks:
+<ol>
+<li>
+Without the use of JavaScript, some browsers require that the contents of the iframe are stored in a <em>separate HTML document</em> rather than being part of the current document. This necessitates a new document composition paradigm that’s very different from the methods currently employed by developers.
+</li>
+<li>
+In the majority of browsers, the contents of an iframe do not natively affect the dimensions of the iframe itself. The only known exception to this at the time of writing is iOS Safari. The contents of a container that is the subject of a Container Query are still expected to be able to affect the layout of its parents in the <em>opposite</em> dimension of the query. For example, in the case of a Container Query that is based on a containers <em>width</em>, it is still expected that the container’s contents affect the <em>height</em> of a container, subject to any overflow rules.
```

Comment:
"natively affect" is unclear. Its also not clear what we mean by Safari behaving differently.

Reply:
Ok, I can expand on that. @tomhodgins do you have any resources or evidence for the Safari iOS thing?

Reply:
There isn't yet a standards-based way to declaratively tell an iframe to auto-size to fit its contents (see https://github.com/w3c/csswg-drafts/issues/1771). Can we leave it at that, rather than getting into implementation-specific hacks?

Reply:
That would be my preference. @tomhodgins ? @eeeps was the intention to include the link to the draft spec you cited?

Reply:
It's been happening on iOs as long as I can remember, I looked it up the first time I saw it and found articles, but lately I can't find anything on Google - so I wrote my own test:

```html
<meta name=viewport content="width=device-width, initial-scale=1">
<iframe></iframe>
<iframe src="http://staticresource.com/"></iframe>
<iframe src="http://eqcss.com/"></iframe>
```

Screenshots: https://imgur.com/a/SKjp4

Reply:
Ok so maybe:

> There currently is no standard, declarative way for an `iframe` to auto-size to fit its contents.
>
> [Note:](https://tabatkins.github.io/bikeshed/#notes-etc) Mobile Safari auto-sizes `<iframe>`s by default, and there is [active discussion](https://github.com/w3c/csswg-drafts/issues/1771) within the CSSWG about whether or not some form of iframe auto-sizing functionality should be standardized.
>
> The contents of a container that is the subject of a Container Query...

Reply:
That is, only if, @tomhodgins, you think Mobile Safari's unique behavior is important to note. I favor keeping the text clean and the argument – iframes weren't built for this and aren't suited for this – simple/strong, but can see why you would want to include the fact that using them is sometimes possible in limited contexts.

Reply:
I think it's worth mentioning that (in addition to other drawbacks of iframes) due to cross-browser inconsistencies with iframe sizing, using iframes isn't a good solution for embedding self-responsive layouts in other documents. To use it reliably you'd have to assign a height to the document or iframe, and in doing so you limit the kinds of layouts you can embed.

I don't know that saying which browsers act differently is important to that point, so we could probably leave browser names out, but if we can dig up a link where they can read more about it that would be helpful for those who want to learn more about why that's true.

Reply:
I’d opt for more generic language here, personally. If only because it’s possible that if a specific (non-standard) implementation might change its behavior in the future.

Reply:
@eeeps @tomhodgins @beep - sorry, dropped out for a while.
To sum, I’m going to change this:

> In the majority of browsers ... iOS Safari.

To this:

> There currently is no standard, declarative way for an iframe to auto-size to fit its contents.

And leave it at that?

Reply:
@andykirk That sounds great to me.

Reply:
@andykirk Me too.

Reply:
Ok, I updated this.
I also re-worked the example/requirements but. I know we still need to discuss that, but I think it reads better now. Happy to change back or re-work if needed.